### PR TITLE
TIP-613: take ES errors into accound during indexation

### PR DESCRIPTION
**Pull request by jjanvier**
During indexation, we didn't take into account ES errors that can occur.
There are 2 types of errors:
- sometimes an exception is thrown by the official ES PHP client 
- sometimes there is just a field `error` set in the response

Errors can occur for instance when we have reach the `index.mapping.total_fields.limit` parameter of Elasticsearch.

Before this PR, the indexation would silently fail. That means, you think that everything's working, but actually, your products or product models are not indexed (and thus are not available in the datagrid)

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | 
| Added integration tests           | 
| Changelog updated                 | 
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | 
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Bundle/ElasticsearchBundle/spec/ClientSpec.php [hunk 1]
```diff
@@ -181,6 +209,43 @@ function it_indexes_with_bulk_several_documents($client)
         $this->bulkIndexes('an_index_type', $documents, 'identifier', Refresh::waitFor());
     }
 
+    public function it_triggers_an_exception_during_the_indexation_of_a_several_documents($client)
```
Comment (SamirBoulil) [inline src/Akeneo/Bundle/ElasticsearchBundle/spec/ClientSpec.php]: it_throws_an_exception_during_the_indexation_of_several_documents

File: src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php [hunk 2]
```diff
@@ -127,6 +127,7 @@ protected function resetElasticsearchIndex(OutputInterface $output)
         $output->writeln('<info>Reset elasticsearch indexes</info>');
 
         $this->getContainer()->get('akeneo_elasticsearch.client.product')->resetIndex();
+        $this->getContainer()->get('akeneo_elasticsearch.client.product_model')->resetIndex();
```
Comment (SamirBoulil) [inline src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: 👍

Comment (nidup) [reply src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: :+1:

Comment (ahocquard) [reply src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: to do in fixture loader too no ? (seems not done)

Comment (jjanvier) [reply src/Pim/Bundle/InstallerBundle/Command/DatabaseCommand.php]: @ahocquard yep, I got a post it for that
but I don't want to struggle more than needed with the CI tonight ^^

**Comment by jjanvier** [comment]
only `navigate_among_variant_entities.feature`* fail

**Pull request merged**